### ISSUE TITLE
The delete task cannot be cancelled

### ISSUE THREAD

**Issue by 125964590** [AliyunContainerService/kubernetes-cronhpa-controller#52]
# The delete task cannot be cancelled
I create 'cronhpa' object and modify the down time but the time before modification also triggers the down event.

## Log

> the time modify before : 0 35 15 * * *
> after : 0 37 15 * * *

```
E0424 15:35:00.753390       1 memcache.go:199] couldn't get resource list for custom.metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0424 15:35:00.853143       1 memcache.go:111] couldn't get resource list for custom.metrics.k8s.io/v1beta1: the server is currently unable to handle the request
time="2020-04-24T15:35:00+08:00" level=info msg="&{{CronHorizontalPodAutoscaler autoscaling.alibabacloud.com/v1beta1} {biz-health-cronhpa-mwa52yrj-cronhpa  test /apis/autoscaling.alibabacloud.com/v1beta1/namespaces/test/cronhorizontalpodautoscalers/biz-health-cronhpa-mwa52yrj-cronhpa f2280e02-85f7-11ea-aa24-00163e30bcfc 285474206 1 2020-04-24 14:50:55 +0800 CST <nil> <nil> map[account_id:paas-176 deployment:biz-health] map[createTime:1587711055430 name:time-test] [] nil []  []} {[] {apps/v1beta2 Deployment biz-health-cronhpa-mwa52yrj} [{up 0 31 15 * * * false 1} {down 0 37 15 * * * false 0}]} {{apps/v1beta2 Deployment biz-health-cronhpa-mwa52yrj} [] [{up c2d6df93-f3dd-4a64-b97d-1e3f22e43207 0 31 15 * * * 1 false Submitted 2020-04-24 15:34:36 +0800 CST } {down b4f7fef5-c742-41cf-92cc-0391270967c3 0 35 15 * * * 0 false Succeed 2020-04-24 15:35:00 +0800 CST cron hpa job down executed successfully. current replicas:1, desired replicas:0}]}} is handled by cron-hpa controller"
time="2020-04-24T15:35:00+08:00" level=info msg="&{{CronHorizontalPodAutoscaler autoscaling.alibabacloud.com/v1beta1} {biz-health-cronhpa-mwa52yrj-cronhpa  test /apis/autoscaling.alibabacloud.com/v1beta1/namespaces/test/cronhorizontalpodautoscalers/biz-health-cronhpa-mwa52yrj-cronhpa f2280e02-85f7-11ea-aa24-00163e30bcfc 285474207 1 2020-04-24 14:50:55 +0800 CST <nil> <nil> map[account_id:paas-176 deployment:biz-health] map[name:time-test createTime:1587711055430] [] nil []  []} {[] {apps/v1beta2 Deployment biz-health-cronhpa-mwa52yrj} [{up 0 31 15 * * * false 1} {down 0 37 15 * * * false 0}]} {{apps/v1beta2 Deployment biz-health-cronhpa-mwa52yrj} [] [{up c2d6df93-f3dd-4a64-b97d-1e3f22e43207 0 31 15 * * * 1 false Submitted 2020-04-24 15:34:36 +0800 CST } {down 6885304e-cf86-4a76-8725-f8a9907e1405 0 37 15 * * * 0 false Submitted 2020-04-24 15:35:00 +0800 CST }]}} has been handled completely."
time="2020-04-24T15:35:00+08:00" level=info msg="&{{CronHorizontalPodAutoscaler autoscaling.alibabacloud.com/v1beta1} {biz-health-cronhpa-mwa52yrj-cronhpa  test /apis/autoscaling.alibabacloud.com/v1beta1/namespaces/test/cronhorizontalpodautoscalers/biz-health-cronhpa-mwa52yrj-cronhpa f2280e02-85f7-11ea-aa24-00163e30bcfc 285474207 1 2020-04-24 14:50:55 +0800 CST <nil> <nil> map[account_id:paas-176 deployment:biz-health] map[name:time-test createTime:1587711055430] [] nil []  []} {[] {apps/v1beta2 Deployment biz-health-cronhpa-mwa52yrj} [{up 0 31 15 * * * false 1} {down 0 37 15 * * * false 0}]} {{apps/v1beta2 Deployment biz-health-cronhpa-mwa52yrj} [] [{up c2d6df93-f3dd-4a64-b97d-1e3f22e43207 0 31 15 * * * 1 false Submitted 2020-04-24 15:34:36 +0800 CST } {down 6885304e-cf86-4a76-8725-f8a9907e1405 0 37 15 * * * 0 false Submitted 2020-04-24 15:35:00 +0800 CST }]}} is handled by cron-hpa controller"
time="2020-04-24T15:35:00+08:00" level=info msg="&{{CronHorizontalPodAutoscaler autoscaling.alibabacloud.com/v1beta1} {biz-health-cronhpa-mwa52yrj-cronhpa  test /apis/autoscaling.alibabacloud.com/v1beta1/namespaces/test/cronhorizontalpodautoscalers/biz-health-cronhpa-mwa52yrj-cronhpa f2280e02-85f7-11ea-aa24-00163e30bcfc 285474207 1 2020-04-24 14:50:55 +0800 CST <nil> <nil> map[account_id:paas-176 deployment:biz-health] map[createTime:1587711055430 name:time-test] [] nil []  []} {[] {apps/v1beta2 Deployment biz-health-cronhpa-mwa52yrj} [{up 0 31 15 * * * false 1} {down 0 37 15 * * * false 0}]} {{apps/v1beta2 Deployment biz-health-cronhpa-mwa52yrj} [] [{up c2d6df93-f3dd-4a64-b97d-1e3f22e43207 0 31 15 * * * 1 false Submitted 2020-04-24 15:34:36 +0800 CST } {down 6885304e-cf86-4a76-8725-f8a9907e1405 0 37 15 * * * 0 false Submitted 2020-04-24 15:35:00 +0800 CST }]}} has been handled completely."
E0424 15:37:00.753412       1 memcache.go:199] couldn't get resource list for custom.metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0424 15:37:00.853525       1 memcache.go:111] couldn't get resource list for custom.metrics.k8s.io/v1beta1: the server is currently unable to handle the request
time="2020-04-24T15:37:00+08:00" level=info msg="&{{CronHorizontalPodAutoscaler autoscaling.alibabacloud.com/v1beta1} {biz-health-cronhpa-mwa52yrj-cronhpa  test /apis/autoscaling.alibabacloud.com/v1beta1/namespaces/test/cronhorizontalpodautoscalers/biz-health-cronhpa-mwa52yrj-cronhpa f2280e02-85f7-11ea-aa24-00163e30bcfc 285474779 1 2020-04-24 14:50:55 +0800 CST <nil> <nil> map[account_id:paas-176 deployment:biz-health] map[createTime:1587711055430 name:time-test] [] nil []  []} {[] {apps/v1beta2 Deployment biz-health-cronhpa-mwa52yrj} [{up 0 31 15 * * * false 1} {down 0 37 15 * * * false 0}]} {{apps/v1beta2 Deployment biz-health-cronhpa-mwa52yrj} [] [{up c2d6df93-f3dd-4a64-b97d-1e3f22e43207 0 31 15 * * * 1 false Submitted 2020-04-24 15:34:36 +0800 CST } {down 6885304e-cf86-4a76-8725-f8a9907e1405 0 37 15 * * * 0 false Succeed 2020-04-24 15:37:00 +0800 CST cron hpa job down executed successfully. current replicas:0, desired replicas:0}]}} is handled by cron-hpa controller"
time="2020-04-24T15:37:00+08:00" level=info msg="&{{CronHorizontalPodAutoscaler autoscaling.alibabacloud.com/v1beta1} {biz-health-cronhpa-mwa52yrj-cronhpa  test /apis/autoscaling.alibabacloud.com/v1beta1/namespaces/test/cronhorizontalpodautoscalers/biz-health-cronhpa-mwa52yrj-cronhpa f2280e02-85f7-11ea-aa24-00163e30bcfc 285474779 1 2020-04-24 14:50:55 +0800 CST <nil> <nil> map[account_id:paas-176 deployment:biz-health] map[createTime:1587711055430 name:time-test] [] nil []  []} {[] {apps/v1beta2 Deployment biz-health-cronhpa-mwa52yrj} [{up 0 31 15 * * * false 1} {down 0 37 15 * * * false 0}]} {{apps/v1beta2 Deployment biz-health-cronhpa-mwa52yrj} [] [{up c2d6df93-f3dd-4a64-b97d-1e3f22e43207 0 31 15 * * * 1 false Submitted 2020-04-24 15:34:36 +0800 CST } {down 6885304e-cf86-4a76-8725-f8a9907e1405 0 37 15 * * * 0 false Succeed 2020-04-24 15:37:00 +0800 CST cron hpa job down executed successfully. current replicas:0, desired replicas:0}]}} has been handled completely."
time="2020-04-24T15:37:00+08:00" level=error msg="Failed to update cronHPA job:down namespace:test cronHPA:biz-health-cronhpa-mwa52yrj-cronhpa,because of Operation cannot be fulfilled on cronhorizontalpodautoscalers.autoscaling.alibabacloud.com \"biz-health-cronhpa-mwa52yrj-cronhpa\": the object has been modified; please apply your changes to the latest version and try again"
```

**Comment by ringtail** [comment]
@NUCsimple Pls check this issue

**Comment by NUCsimple** [comment]
yeah,It's true.During the test, I also encountered this problem. As shown in the picture below, I changed the shrinking time from the first second of every minute to 30 second, but in fact, the first second of every minute was also reduced.
![image](https://user-images.githubusercontent.com/37863411/80207775-60128e80-8661-11ea-923b-0326db0e9778.png)
![image](https://user-images.githubusercontent.com/37863411/80207830-7f112080-8661-11ea-852f-31eac8d3663d.png)

**Comment by ringtail** [comment]
@NUCsimple Pls drive the bug fix.

**Comment by NUCsimple** [comment]
@ringtail .Sorry for that test.I retested again.The first,I set the scale down time is the  35s per minute.Like the picture.So,I think,there should be no problem with this component. I'm sorry for the confusion with you.
![image](https://user-images.githubusercontent.com/37863411/80303847-5fa30080-87e5-11ea-90ce-1f8f26a18a03.png)
![image](https://user-images.githubusercontent.com/37863411/80303872-84977380-87e5-11ea-86e2-49737e0c401a.png)
then;In the next cycle, I set this time to 45s,you can see that, in the 35s ,the pod is not scale down.
![image](https://user-images.githubusercontent.com/37863411/80304029-3fc00c80-87e6-11ea-80f1-6e58037cdab6.png)
@125964590 If there is still a problem in your test, can you attach the picture to the attachment so that we can better observe this phenomenon.

**Comment by 125964590** [comment]
I change the cronhpa object through http request.

```java
  public CronHpa replaceCronHpa(String name, String namespace, CronHpa body, String pretty)
      throws ApiException {
    ApiResponse<CronHpa> resp = replaceCronHpaWithHttpInfo(name, namespace, body, pretty);
    return resp.getData();
  }

  public ApiResponse<CronHpa> replaceCronHpaWithHttpInfo(
      String name, String namespace, CronHpa body, String pretty) throws ApiException {
    com.squareup.okhttp.Call call =
        replaceCronHpaValidateBeforeCall(name, namespace, body, pretty, null, null);
    Type localVarReturnType = new TypeToken<CronHpa>() {}.getType();
    return apiClient.execute(call, localVarReturnType);
  }

  private com.squareup.okhttp.Call replaceCronHpaValidateBeforeCall(
      String name,
      String namespace,
      CronHpa body,
      String pretty,
      final ProgressResponseBody.ProgressListener progressListener,
      final ProgressRequestBody.ProgressRequestListener progressRequestListener)
      throws ApiException {

    // verify the required parameter 'name' is set
    if (name == null) {
      throw new ApiException(
          "Missing the required parameter 'name' when calling replaceCronHpa(Async)");
    }

    // verify the required parameter 'body' is set
    if (body == null) {
      throw new ApiException(
          "Missing the required parameter 'body' when calling replaceCronHpa(Async)");
    }

    com.squareup.okhttp.Call call =
        replaceCronHpaCall(
            name, namespace, body, pretty, progressListener, progressRequestListener);
    return call;
  }

  public com.squareup.okhttp.Call replaceCronHpaCall(
      String name,
      String namespace,
      CronHpa body,
      String pretty,
      final ProgressResponseBody.ProgressListener progressListener,
      final ProgressRequestBody.ProgressRequestListener progressRequestListener)
      throws ApiException {
    Object localVarPostBody = body;

    // create path and map variables
    String localVarPath =
        "/apis/autoscaling.alibabacloud.com/v1beta1/namespaces/{namespace}/cronhorizontalpodautoscalers/{name}"
            .replaceAll("\\{" + "namespace" + "\\}", namespace)
            .replaceAll("\\{" + "name" + "\\}", apiClient.escapeString(name.toString()));

    List<Pair> localVarQueryParams = new ArrayList<Pair>();
    List<Pair> localVarCollectionQueryParams = new ArrayList<Pair>();
    if (pretty != null) {
      localVarQueryParams.addAll(apiClient.parameterToPair("pretty", pretty));
    }

    Map<String, String> localVarHeaderParams = new HashMap<String, String>();

    Map<String, Object> localVarFormParams = new HashMap<String, Object>();

    final String[] localVarAccepts = {
      "application/json", "application/yaml", "application/vnd.kubernetes.protobuf"
    };
    final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);
    if (localVarAccept != null) localVarHeaderParams.put("Accept", localVarAccept);

    final String[] localVarContentTypes = {"*/*"};
    final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);
    localVarHeaderParams.put("Content-Type", localVarContentType);

    if (progressListener != null) {
      apiClient
          .getHttpClient()
          .networkInterceptors()
          .add(
              new com.squareup.okhttp.Interceptor() {
                @Override
                public com.squareup.okhttp.Response intercept(
                    com.squareup.okhttp.Interceptor.Chain chain) throws IOException {
                  com.squareup.okhttp.Response originalResponse = chain.proceed(chain.request());
                  return originalResponse
                      .newBuilder()
                      .body(new ProgressResponseBody(originalResponse.body(), progressListener))
                      .build();
                }
              });
    }

    String[] localVarAuthNames = new String[] {"BearerToken"};
    return apiClient.buildCall(
        localVarPath,
        "PUT",
        localVarQueryParams,
        localVarCollectionQueryParams,
        localVarPostBody,
        localVarHeaderParams,
        localVarFormParams,
        localVarAuthNames,
        progressRequestListener);
  }
```

**Comment by ringtail** [comment]
```
time="2020-04-24T15:37:00+08:00" level=error msg="Failed to update cronHPA job:down namespace:test cronHPA:biz-health-cronhpa-mwa52yrj-cronhpa,because of Operation cannot be fulfilled on cronhorizontalpodautoscalers.autoscaling.alibabacloud.com \"biz-health-cronhpa-mwa52yrj-cronhpa\": the object has been modified; please apply your changes to the latest version and try again"
```
You can try to update the object with revision.

**Comment by ringtail** [comment]
Fix due to #63
